src/rules/no_empty_enum.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020-2021 the Deno authors. All rights reserved. MIT license.
+use super::{Context, LintRule, DUMMY_NODE};
+use crate::ProgramRef;
+use deno_ast::swc::ast::TsEnumDecl;
+use deno_ast::swc::visit::Node;
+use deno_ast::swc::visit::Visit;
+
+#[derive(Debug)]
+pub struct NoEmptyEnum;
+
+const CODE: &str = "no-empty-enum";
+const MESSAGE: &str = "An empty enum is equivalent to `{{}}`.";
+const HINT: &str = "Remove this enum or add members to this enum.";
+
+impl LintRule for NoEmptyEnum {
+  fn new() -> Box<Self> {
+    Box::new(NoEmptyEnum)
+  }
+
+  fn tags(&self) -> &'static [&'static str] {
+    &["recommended"]
+  }
+
+  fn code(&self) -> &'static str {
+    CODE
+  }
+
+  fn lint_program<'view>(
+    &self,
+    context: &mut Context<'view>,
+    program: ProgramRef<'view>,
+  ) {
+    let mut visitor = NoEmptyEnumVisitor::new(context);
+    match program {
+      ProgramRef::Module(m) => visitor.visit_module(m, &DUMMY_NODE),
+      ProgramRef::Script(s) => visitor.visit_script(s, &DUMMY_NODE),
+    }
+  }
+
+  #[cfg(feature = "docs")]
+  fn docs(&self) -> &'static str {
+    include_str!("../../docs/rules/no_empty_enum.md")
+  }
+}
+
+struct NoEmptyEnumVisitor<'c, 'view> {
+  context: &'c mut Context<'view>,
+}
+
+impl<'c, 'view> NoEmptyEnumVisitor<'c, 'view> {
+  fn new(context: &'c mut Context<'view>) -> Self {
+    Self { context }
+  }
+}
+
+impl<'c, 'view> Visit for NoEmptyEnumVisitor<'c, 'view> {
+  fn visit_ts_enum_decl(
+    &mut self,
+    enum_decl: &TsEnumDecl,
+    _parent: &dyn Node,
+  ) {
+    if enum_decl.members.is_empty() {
+      self.context.add_diagnostic_with_hint(
+        enum_decl.span,
+        CODE,
+        MESSAGE,
+        HINT,
+      );
+    }
+  }
+}
```

Comment:
If we use `Visit` trait, we have to explicitly call a method to traverse AST nodes further. This means that the current code won't capture the error in the following example:

```ts
enum Foo {
  One = 1,
  Two = (() => {
    enum Bar {} // this won't be detected!
    return 42;
  })(),
}
```

Although this is a valid TypeScript, of course people might not write such a tricky code, but it should be better for the linter to detect it as well.
To address this issue, I recommend to use `VisitAll` trait instead. With this trait, we won't need to do something explicitly, the visitor will traverse all the nodes automatically.
(Or you can also use `Handler` trait - if you're interested in it, see the concrete example, like https://github.com/denoland/deno_lint/blob/9669c247be0e6ce4fcd376acdfd389251e83ed8a/src/rules/no_with.rs)

Reply:
I can't use ```VisitAll```, it doesn't work, so I change to ```Handler``` but added if statement, could you review again plz @magurotuna and added that example to test (I've never seen that tricky code before and can't imagine)

Reply:
And ci failed on format step 😥

Reply:
Now the implementation with `Handler` looks perfect 👍 

> And ci failed on format step 😥

We have a script for formatting the code. So just running `./tools/format.ts` will help you.